### DISCUSSION TITLE
Reverse ORDER BY

### DISCUSSION THREAD

**Discussion by ziima** [sqlalchemy/sqlalchemy#10361] [Usage Questions]
I run into a case where I need to sometimes reverse the order in `ORDER BY` clause. Is there a way to do it easily?

I tried `~order_by_clause`, but it returns `'NOT my_column DESC'`. Should the negative operator reverse the `ORDER BY` clause?

**Comment by zzzeek** [comment]
column expressions have [.desc()](https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.Column.desc) and .asc() for this


    stmt = select(table).order_by(table.c.col.desc())

**Reply by ziima** [reply]
Yes, but my problem is that I already have an ordered query, which I need to reverse. I don't know whether it's ascending or descending at the moment.

**Comment by CaselIT** [comment]
>  Should the negative operator reverse the ORDER BY clause?

No, since it's perfectly legal to order by `not column` and it would be surprising if that were to render `column desc`

**Reply by ziima** [reply]
Of course, I didn't realize that.
